### PR TITLE
[SRVKE-1479] Add docs on Kafka broker's templates config

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
`serverless-docs-1.32`+

Issue:
https://issues.redhat.com/browse/SRVKE-1479

Link to docs preview:
https://72688--ocpdocs-pr.netlify.app/openshift-serverless/latest/eventing/tuning/overriding-config-eventing#knative-eventing-modifying-consumer-group-IDs-and-topic-names_overriding-config-eventing

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by Cali0707** [comment]
@rh-max maybe it's worth mentioning that the values that are available in the templates are from the broker/trigger/channel `ObjectMeta`?

cc @pierDipi

File: modules/serverless-kafka-broker-configmap.adoc
Comment (pierDipi) [inline modules/serverless-kafka-broker-configmap.adoc:65]: This is not the right configmap for this feature, @Cali0707 isn't it in `config-kafka-features`?

Comment (pierDipi) [reply modules/serverless-kafka-broker-configmap.adoc:65]: To be precise, this needs to be configured on the KnativeKafka resource by an "administrator"

Comment (Cali0707) [reply modules/serverless-kafka-broker-configmap.adoc:65]: yep this is in `config-kafka-features` @pierDipi - good catch, thanks!

Comment (rh-max) [reply modules/serverless-kafka-broker-configmap.adoc:88]: @pierDipi Thanks, fixed

**Review by kcarmichael08 — COMMENTED** [review]
LGTM! Not sure if the serverless repo requires commits be squashed but if so you may need to do that before merging.

File: modules/serverless-kafka-broker-configmap.adoc
Comment (pierDipi) [inline modules/serverless-kafka-broker-configmap.adoc:59]:
```suggestion
. To change templates for generating consumer group IDs and topic names used by your triggers, brokers, and channels, modify the `KnativeKafka` resource:
```

Comment (pierDipi) [reply modules/serverless-kafka-broker-configmap.adoc:59]: Users need to configure the templates on the `KnativeKafka` resource in the `knative-eventing` namespace as the ConfigMap is managed by the operator:

```yaml
# ...
kind: KnativeKafka
metadata:
  name: <..>
  namespace: knative-eventing
# ...

spec:
  config:
    config-kafka-features:
       triggers-consumergroup-template: knative-trigger-{{ .Namespace }}-{{ .Name }}         # <-- change template value here
```

**Review by pierDipi — COMMENTED** [review]
@rh-max since this is configuration is for admins vs dev, the page we're editing is meant for devs and it's not related to the goal of the page, I would place this content under "Tuning eventing configuration" https://72688--ocpdocs-pr.netlify.app/openshift-serverless/latest/eventing/tuning

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (mgencur) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:27]: This needs to be `name: knative-kafka` . It's not a name of some custom configmap. Should be similar to https://docs.openshift.com/serverless/1.31/install/kube-rbac-proxy-kafka.html#serverless-configuring-kube-rbac-proxy-resources-for-kafka_kube-rbac-proxy-kafka

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (mgencur) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:48]: See my previous comment. Needs to be `name: knative-kafka`

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (mgencur) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:59]: It's not a config map. It's a custom resource named KnativeKafka.

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (mgencur) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:63]: Not a config_map_filename. Probalby better to mention `<knative_kafka_filename>`

**Comment by rh-max** [comment]
/label merge-review-needed
To the merge reviewer: please do not merge on completion, we are waiting for the GA confirmation. Thank you.

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (pierDipi) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:56]: There is a render issue

image

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (pierDipi) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:55]:
```suggestion
      triggers.consumergroup.template: "{% raw %}"knative-trigger-{{ .Namespace }}-{{ .Name }}-{{ .annotations.my-annotation }}"{% endraw %}"
      brokers.topic.template: "{% raw %}"knative-broker-{{ .Namespace }}-{{ .Name }}-{{ .annotations.my-annotation }}"{% endraw %}"
      channels.topic.template: "{% raw %}"messaging-kafka.{{ .Namespace }}.{{ .Name }}-{{ .annotations.my-annotation }}"{% endraw %}"
```

File: modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc
Comment (adellape) [inline modules/knative-eventing-modifying-consumer-group-IDs-and-topic-names.adoc:1]: Not blocking merge on this, but filenames should be all lowercase. FWIW, Git is sometimes difficult to wrangle to after the fact to replace in-place.

**Review by adellape — APPROVED** [review]
Confirmed with Max via Slack that this is ready for merge.

**Pull request merged**
